### PR TITLE
feat: + tree command

### PR DESCRIPTION
**Pull request by iorisa**
**Features**
Implement the same functionality as the `tree` command.
Example:
```
        root
        +-- dir1
        |   +-- file1.txt
        |   +-- file2.txt
        +-- dir2
        |   +-- subdir1
        |   |   +-- file1.txt
        |   |   +-- file2.txt
        |   +-- subdir2
        |       +-- file1.txt
        |       +-- file2.txt
        +-- file.txt
```

### CODE REVIEW

File: metagpt/utils/tree.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        |       +-- file1.txt
+        |       +-- file2.txt
+        +-- file.txt
+"""
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Callable, Dict, List
+
+from anthropic import BaseModel
```
Comment (geekan) [inline metagpt/utils/tree.py:27]: ?

File: metagpt/utils/tree.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+class Tree(BaseModel):
+    """
+    Represents a directory tree structure.
+
+    Attributes:
+        root (str): The root directory of the tree.
+        tree (Dict[str, Dict]): The tree structure as a dictionary.
+
+    Methods:
+        print: Print the directory tree structure.
```
Comment (geekan) [inline metagpt/utils/tree.py:40]: give examples

File: metagpt/utils/tree.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    """
+
+    root: str
+    tree: Dict[str, Dict] = Field(default_factory=dict)
```
Comment (geekan) [inline metagpt/utils/tree.py:45]: Set as private variable

**Comment by geekan** [comment]
I still want to ask a question: now that the tree command exists and it provides a large number of options, why should we reimplement it? What are the benefits?

File: metagpt/utils/tree.py (hunk 1 above)
Comment (iorisa) [reply metagpt/utils/tree.py]: Fixed.
Classes have been simplified into functions.

File: metagpt/utils/tree.py (hunk 2 above)
Comment (iorisa) [reply metagpt/utils/tree.py]: done

File: metagpt/utils/tree.py (hunk 3 above)
Comment (iorisa) [reply metagpt/utils/tree.py]: Fixed.
Classes have been simplified into functions

**Review by geekan — APPROVED** [review]

**Pull request merged**